### PR TITLE
Add: make target to run black-formatter as it is used in PR checks.

### PR DESCRIPTION
**Pull request by frosch123**
I don't know how you invoke black. But there are quite some custom options for NML.

### CODE REVIEW

**Comment by FLHerne** [comment]
Comment as on IRC:

This should probably also add a `--check` run of black, and flake8, to `make test`.

Potentially we could then drop black and flake8 as separate GitHub actions because the main testing would run them, but then we'd lose inline notes from TrueBrain's action.

**Comment by frosch123** [comment]
I added them to "make test".
But now "make test" depends on "flake8" and "black" being installed.

**Comment by FLHerne** [comment]
I don't think that's an issue really; `black` at least you pretty much need to make any significant changes without upsetting the commit-checker.

Perhaps the people building distro packages will mind?

In principle we could have a test script that checks whether they're installed and warns but succeeds otherwise, but that seems TMWFTLB.

File: Makefile [hunk 1]
```diff
@@ -15,3 +17,10 @@ extensions:
 
 clean:
 	$(MAKE) -C regression clean
+
+flake:
+	$(PYTHON) -m black -check $(BLACK_OPTIONS) nml
```
Comment (FLHerne) [inline Makefile:22]: Should be `--check` with two `-`s.

**Comment by FLHerne** [comment]
I guess this should add `black` and `flake8` to "Optional Dependencies" of README.md.

File: README.md [hunk 2]
```diff
@@ -50,6 +50,12 @@ To install NML you'll need these 3rd party packages:
 - gcc (or possibly another c++ compiler).
   Needed to compile the cython version of the lz77 module for grf encoding.
 
+To improve NML and submit pull requests you'll need these code formaters and checkers.
```
Comment (FLHerne) [inline README.md:53]: [pedant] "formatters".

Also this seems a bit vague. Non-contributors who want to run `make test` (e.g. distro packagers) will also need these, and they aren't strictly *required* for making a PR.

 Perhaps something like:

> Running `make test` requires these code formatters and checkers. You don't need these to use NML, only to test or help improve it.

**Comment by LordAro** [comment]
Not sure I like this being rolled into the `make test` rule. A separate "checkers" rule, perhaps?

File: Makefile [hunk 3]
```diff
@@ -1,11 +1,13 @@
 MAKE?=make
 PYTHON?=/usr/bin/env python3
+BLACK_OPTIONS=-l 120 --exclude 'action2var_variables.py|action3_callbacks.py|generated'
```
Comment (TrueBrain) [inline Makefile:3]: if you like, black options can also be put in `pyproject.toml`, see https://black.readthedocs.io/en/stable/pyproject_toml.html .
This makes sure that if someone executed `black`, it picks up the right rules too.
I have yet to apply this for all other OpenTTD projects :)

File: Makefile [hunk 4]
```diff
@@ -15,3 +17,10 @@ extensions:
 
 clean:
 	$(MAKE) -C regression clean
+
+flake:
```
Comment (TrueBrain) [inline Makefile:21]: My OCD cannot handle that `make flake` also executes black .. but using any other name is silly :P

**Review by TrueBrain — APPROVED** [review]
I think such quality of life improvements are always a good thing. You cannot execute tests often enough.

**Comment by TrueBrain** [comment]
> 
> 
> Not sure I like this being rolled into the `make test` rule. A separate "check" rule, perhaps?

As much as a regression shows a runtime error, `flake` shows possible runtime errors. One can debate if `black` is really a test, but flake for sure is. And with having `flake` under `make test`, you might as well add `black` too, as .. why not! :D

**Comment by frosch123** [comment]
Conclusion: noone likes pyproject.toml

**Pull request merged**